tests/e2e/main_test.go

```diff
@@ -277,3 +277,42 @@ func TestFailedRuleEvaluations(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestGrafana(t *testing.T){
+	kClient := promClient.kubeClient
+	grafanaScale, err := kClient.AppsV1().Deployments("monitoring").GetScale(context.Background(), "grafana", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	
+	// Loop for 5 mins
+	for x := 0; x < 10; x++ {
+		podList, err := kClient.CoreV1().Pods("monitoring").List(context.Background(), metav1.ListOptions{LabelSelector: grafanaScale.Status.Selector})
+		if err != nil {
+			t.Fatal(err)
+		}
+		
+		err = func() error {
+			if len(podList.Items) != int(grafanaScale.Spec.Replicas) {
```

Comment:
In the podList, you might get pods that are created but not ready so even if there is a problem in the configuration we might not see it. I think it would be better to compare `grafanaScale.Spec.Replicas` to `grafanaScale.Spec.ReadyReplicas`.